File: listas/lista5/f.cpp

using namespace std;
#define endl '\n'


int euclides(int a, int b) {
	if (b > a) swap(a, b);
	while(b != 0) {
		int resto = a % b;
		a = b;
		b = resto;
	}

	return a;
}


void solve() {
	int n; cin >> n;
	vector<int> nums(n);

	for(int i = 0; i < n; i++) {
		cin >> nums[i];
	}

	int mdc = nums[0];

	for(int i = 1; i < n; i++) {
		mdc = euclides(mdc, nums[i]);
	}

	int cont = 0;

	// HAHAHAHHAHAHAHAHHAHAHAHAHAHAHHAHAHAHAHAHHAHAHAHAHHAHAHAHAHAHAHHAHA

	for(int i = 1; i * i <= mdc; i++) {
		if(mdc % i == 0) {
			if(i * i == mdc) {
				cont++;
			} else {
				cont += 2;
			}
		}
	}

	cout << cont<< endl;
}


int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);
	solve();
    return 0;
}

File: listas/lista5/f.py
def euclides(a, b):
    while b!= 0:
        a, b = b, a % b

    return a

n = int(input())
nums = list(map(int, input().split()))
nums.sort()

mdc = nums[0]

for i in range(1, n):
    mdc = euclides(mdc, nums[i]);

target = int(mdc ** 0.5)

ok = False

if(target == mdc ** 0.5):
    ok = True

cont = 0
for i in range(1, target + 1):
    if mdc % i == 0:
        cont += 1

if ok:
    print(2 * cont -1)
else:
    print(2 * cont)
